stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpUtil.java

```diff
@@ -377,10 +377,23 @@ public static String getSessionID(String cookieHeader) {
                 .findFirst().get().trim().substring(Constants.SESSION_ID.length());
     }
 
-    public static void addHTTPSessionAndCorsHeaders(HTTPCarbonMessage requestMsg, HTTPCarbonMessage responseMsg) {
+    public static void addHTTPSessionAndCorsHeaders(Context context, HTTPCarbonMessage requestMsg,
+                                                    HTTPCarbonMessage responseMsg) {
         Session session = (Session) requestMsg.getProperty(Constants.HTTP_SESSION);
         if (session != null) {
-            session.generateSessionHeader(responseMsg);
+            boolean isSecureRequest = false;
+            AnnAttachmentInfo configAnn = context.getServiceInfo().getAnnotationAttachmentInfo(
+                    Constants.PROTOCOL_PACKAGE_HTTP, Constants.ANN_NAME_CONFIG);
+            if (configAnn != null) {
+                AnnAttributeValue httpsPortAttrVal = configAnn.getAttributeValue(Constants.ANN_CONFIG_ATTR_HTTPS_PORT);
+                if (httpsPortAttrVal != null) {
+                    Integer listenerPort = (Integer) requestMsg.getProperty(Constants.LISTENER_PORT);
+                    if (listenerPort != null && httpsPortAttrVal.getIntValue() == listenerPort.intValue()) {
```

Comment:
We can use listenerPort instead of listenerPort.intValue() here?

Reply:
Yes. Corrected. Thanks!